### PR TITLE
upgrade tcgc to 0.36.1 and onboard Decimal

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/3975

- upgrade tcgc to 0.36.1 and cadl-ranch-specs to 0.27.0
- onboard Decimal scalar

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "@azure-tools/typespec-autorest": "0.36.1",
     "@azure-tools/typespec-azure-core": "0.36.0",
     "@azure-tools/typespec-azure-resource-manager": "0.36.1",
-    "@azure-tools/typespec-client-generator-core": "0.36.0",
+    "@azure-tools/typespec-client-generator-core": "0.36.1",
     "@azure-tools/typespec-csharp": "file:src/TypeSpec.Extension/Emitter.Csharp",
```
Comment (pshao25) [inline package.json:11]: Can we try to remove these duplicate version? I don't know why we have this.

File: test/CadlRanchProjects.Tests/type-scalar.cs [hunk 2]
```diff
@@ -51,5 +52,69 @@ public class TypeScalarTests : CadlRanchTestBase
...
+        public Task Type_Scalar_DecimalType_RequestParameter() => Test(async (host) =>
+        {
+            var response = await new ScalarClient(host, null).GetDecimalTypeClient().RequestParameterAsync(0.33333m);
+            Assert.AreEqual(204, response.Status);
+        });
+        [Test]
+        public Task Type_Scalar_Decimal128Type_ResponseBody() => Test(async (host) =>
+        {
+            var response = await new ScalarClient(host, null).GetDecimal128TypeClient().ResponseBodyAsync();
+            Assert.AreEqual(0.33333, response.Value);
```
Comment (pshao25) [inline test/CadlRanchProjects.Tests/type-scalar.cs:77]: So we don't need 0.33333m ?

File: package.json (hunk 1 above)
Comment (chunyu3) [reply package.json:11]: @pshao25 I don't know the purpose of this. Michael add this, maybe for float dependency. I suggest not to update this in this PR.

File: test/CadlRanchProjects.Tests/type-scalar.cs [hunk 3]
```diff
@@ -51,5 +52,69 @@ public class TypeScalarTests : CadlRanchTestBase
             var response = await new ScalarClient(host, null).GetUnknownClient().PutAsync(body);
             Assert.AreEqual(204, response.Status);
         });
+        [Test]
```
[the remark this answers is not in the document]
Comment (chunyu3) [reply test/CadlRanchProjects.Tests/type-scalar.cs:55]: @pshao25 we have tests for type-scalar, and I added tests for property/value-types and projection/projected-name.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**